src/modules/aws_sqs/entity/queue.ts

```diff
@@ -0,0 +1,149 @@
+import { Max, Min } from 'class-validator';
+import { Column, Entity, JoinColumn, ManyToOne, PrimaryGeneratedColumn } from 'typeorm';
+
+import { cloudId } from '../../../services/cloud-id';
+import { AwsRegions } from '../../aws_account/entity';
+
+@Entity()
+export class Queue {
+  /**
+   * @private
+   * Auto-incremented ID field
+   */
+  @PrimaryGeneratedColumn()
+  id?: number;
+
+  /**
+   * @public
+   * Amazon SQS assigns a unique identifier called a queue URL to each new queue.
+   */
+  @Column({
+    nullable: true,
+    type: 'varchar',
+  })
+  @cloudId
+  url?: string;
+
+  /**
+   * @public
+   * Name for the SQS queue
+   * @see https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-sqs/interfaces/createqueuecommandinput.html#queuename
+   */
+  @Column({
+    nullable: false,
+    type: 'varchar',
+  })
+  name: string;
+
+  /**
+   * @public
+   * Designates a queue as FIFO. If not specified, the queue will be created in standard mode.
+   * @see https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-sqs/interfaces/createqueuecommandinput.html
+   */
+  @Column({
+    default: false,
+    type: 'boolean',
+  })
+  fifoQueue: boolean;
+
+  /**
+   * @public
+   * The visibility timeout for the queue, in seconds. Valid values: An integer from 0 to 43,200 (12 hours).
+   * @see https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-sqs/interfaces/createqueuecommandinput.html
+   */
+  @Min(0)
+  @Max(43_200)
+  @Column({
+    default: 30,
+    type: 'integer',
+  })
+  visibilityTimeout: number;
+
+  /**
+   * @public
+   * The length of time, in seconds, for which the delivery of all messages in the queue is delayed.
+   * @see https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-sqs/interfaces/createqueuecommandinput.html
+   */
+  @Min(0)
+  @Max(900)
+  @Column({
+    default: 0,
+    type: 'integer',
+  })
+  delaySeconds: number;
+
+  /**
+   * @public
+   * The length of time, in seconds, for which a ReceiveMessage action waits for a message to arrive.
+   * @see https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-sqs/interfaces/createqueuecommandinput.html
+   */
+  @Min(0)
+  @Max(20)
+  @Column({
+    default: 0,
+    type: 'integer',
+  })
+  receiveMessageWaitTimeSeconds: number;
+
+  /**
+   * @public
+   * The length of time, in seconds, for which Amazon SQS retains a message.
+   * @see https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-sqs/interfaces/createqueuecommandinput.html
+   */
+  @Min(60)
+  @Max(1_209_600)
```

Comment:
Nit: comment on what 1,209,600 seconds means (14 days)

```suggestion
  @Max(1_209_600) // 14 days
```

Reply:
(Also I am reminded of "A Fire Upon The Deep"'s use of Msec, Gsec, etc units once humanity no longer even remembers its home world but Unix somehow survives ;) )